akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -125,14 +138,21 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
         })
 
       def notifySubscribersFor(key: AbstractServiceKey, state: ServiceRegistry): Unit = {
-        val msg = ReceptionistMessages.Listing(key.asServiceKey, state.actorRefsFor(key))
+        // filter tombstoned refs to avoid an extra update
+        // to subscribers in the case of lost removals (because of how ORMultiMap works)
```

Comment:
Don't we always send an update regardless?

Reply:
This is for the case where the local actor died, and was removed from the receptionist, which we got an update about, but then is reintroduced because of concurrent modification to the same service key, but we already know it died, so no need to tell our local subscribers that it is alive again.